### PR TITLE
fix(braintree): give better errors to our users

### PR DESCRIPTION
**Pull request by jefflembeck**
This parses braintree errors given to us by chargify and displays them for our users

### CODE REVIEW

File: www/assets/js/braintree-error-parser.js [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+      this.code = options.code;
+      this.text = options.text;
+      this.implication = options.implication;
+    }
+  }
+
+  class BraintreeErrorParser {
+    static parse(errorText) {
+        var regExp = /\(([^)]+)\)/;
+        var errorCode = regExp.exec(errorText);
```
Comment (stoyan) [inline www/assets/js/braintree-error-parser.js:436]: nit: `const` maybe?

**Review by stoyan — APPROVED** [review]
sweet!

Maybe in a follow up we can make braintree-error-parser.js a module and load on demand in the event listener?

**Review by tkadlec — APPROVED** [review]
LGTM. 

We should definitely work on more user-friendly alternatives to the messaging for some of these, but let's get this out so at least we have more actionable information and then we can massage that after.

**Comment by jefflembeck** [comment]
Yeah, i'm going to fix this up today. Will get this in and will make it nicer.

**Pull request merged**